homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -658,6 +661,33 @@ def __init__(self, name, device, entry, unique_id, coordinator, description):
         self._attr_native_value = self._determine_native_value()
         self._attr_extra_state_attributes = self._extract_attributes(coordinator.data)
 
+    def _extract_value_from_attribute(self, state, attribute):
+        """Overridden to handle different time units for timedelta.
+
+        This ensures that the device reported native values are converted to the ones
+        as defined in the entity descriptions.
+
+        This needs to be done here as the parent class does not have access
+        to the entity description.
+        """
+        value = getattr(state, attribute)
+        if not isinstance(value, timedelta):
+            return super()._extract_value_from_attribute(state, attribute)
+
+        conversion_map = {
+            TIME_SECONDS: timedelta(seconds=1),
+            TIME_MINUTES: timedelta(minutes=1),
+            TIME_HOURS: timedelta(hours=1),
+            TIME_DAYS: timedelta(days=1),
+        }
+
+        native_unit = self.entity_description.native_unit_of_measurement
+        if native_unit in conversion_map:
+            return round((value / conversion_map[native_unit]), 2)
+
+        _LOGGER.debug("No known conversion for %s, returning seconds", native_unit)
```

Comment:
Shouldn't we raise a RuntimeError instead here? We control both the description native unit of measurement and the conversion map so if we get here it should only be due to a developer mistake. Am I missing something?

Reply:
Or maybe we simply ignore checking whether the existing conversion exists and let it bubble up, as that shouldn't really ever happen.